### PR TITLE
fix: Disable auto-zoom on iOS

### LINKED ISSUE #47328: Auto zoom when using Nextcloud on Safari iOS
I used our web interface a couple of times now on mobile safari and if feels quite strange and is partially also annoying that Safari is automatically zooming to input elements and also contenteditables (e.g. in text) when you focus them.

This used to work differently and in my perspective also should for a web app that is not a pure content page.

This was introduced in https://github.com/nextcloud/server/commit/ac736b2f82adfe73176fa78620e617ff17dbc91e as a accessibility fix where I still agree that manually zooming should be possible, however maybe we can get rid of the automatic zoomin.

I found https://stackoverflow.com/questions/2989263/disable-auto-zoom-in-input-text-tag-safari-on-iphone as a possible option.

@nextcloud/designers What do you think about that? Any arguments against changing this?

### PR DESCRIPTION
**Pull request by susnux**
* Resolves https://github.com/nextcloud/server/issues/47328

## Summary

When using iOS and focusing an input element the view should not be zoomed. So if we set a maximum scale iOS will not auto-zoom but still allow users to zoom. But we can not do this by default as this will disable user zoom on Chrome.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

**Review by jancborchardt — APPROVED** [review]

File: lib/private/TemplateLayout.php [hunk 1]
```diff
@@ -286,6 +288,13 @@ public function __construct($renderAs, $appId = '') {
 			}
 		}
 
+		$request = \OCP\Server::get(IRequest::class);
+		if ($request->isUserAgent([Request::USER_AGENT_CLIENT_IOS, Request::USER_AGENT_SAFARI])) {
```
Comment (juliushaertl) [inline lib/private/TemplateLayout.php:292]: Just tested and those two do not match mobile Safari, the following should work adding the following match group `(Mobile\/[0-9.A-Z]+)`, not sure if we want to:

- Include it in USER_AGENT_SAFARI: `Mozilla\/5\.0 \([^)]+\) AppleWebKit\/[0-9.]+ \(KHTML, like Gecko\) Version\/[0-9.]+ (Mobile\/[0-9.A-Z]+)? Safari\/[0-9.A-Z]+`
- Have a separate USER_AGENT_SAFARI_MOBILE: `Mozilla\/5\.0 \([^)]+\) AppleWebKit\/[0-9.]+ \(KHTML, like Gecko\) Version\/[0-9.]+ (Mobile\/[0-9.A-Z]+) Safari\/[0-9.A-Z]+`

Comment (susnux) [reply lib/private/TemplateLayout.php]: Thanks for testing! I chose the second as we also have the mobile chrome variant

**Review by juliusknorr — APPROVED** [review]
Awesome, tested and works now 👍 

Just Ci is not happy yet

**Pull request merged**